### DISCUSSION TITLE
Support stopping / turning off / deactivating debugpy

### DISCUSSION THREAD

**Discussion by yang** [microsoft/debugpy#1790] [Enhancements]
Debugpy imposes a performance impact. Once you start debugpy in a process, the process is forever slow thereafter. So outside of the most performance insensitive of applications, deciding to use debugpy is not an easy decision—you are consigning that process to be killed.

A way to stop or disable debugpy would make it practical to use in many more settings (including production settings).

(Ideally this would be the default behavior—when you listen for debuggers, it doesn't actually do anything to make the program slower. But not sure if I'm missing anything there.)

**Comment by rchiodo** [comment]
Transferring to a discussion item for upvotes. I think with sys.monitoring support we might be able to do this. We'd just disable the monitor. Debugpy would still be in the process but it wouldn't be doing anything. Not zero overhead but minimal.
